### ISSUE TITLE
`error: attribute 'patchedNix' missing` after calling `bud update`

### ISSUE THREAD

**Issue by codygman** [divnix/devos#404]
```
❯ git clone https://github.com/divnix/devos.git
Cloning into 'devos'...
remote: Enumerating objects: 6813, done.
remote: Counting objects: 100% (2052/2052), done.
remote: Compressing objects: 100% (950/950), done.
remote: Total 6813 (delta 1279), reused 1663 (delta 1069), pack-reused 4761
Receiving objects: 100% (6813/6813), 3.68 MiB | 4.27 MiB/s, done.
Resolving deltas: 100% (3816/3816), done.

/tmp on  (us-east-1) 
❯ cd devos
direnv: error /tmp/devos/.envrc is blocked. Run `direnv allow` to approve its content

devos  main on  (us-east-1) 
❯ git status
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean

devos  main on  (us-east-1) 
❯ PAGER=cat git log --oneline -n1
b15b8bf (HEAD -> main, origin/main, origin/HEAD) Do not hardcode UID for the exemplary user

devos  main on  (us-east-1) 
❯ direnv allow
direnv: loading /tmp/devos/.envrc
direnv: using flake
direnv: renewed cache
[git.hooks] found updates
'/tmp/devos/.git/hooks/pre-commit' -> '/nix/store/nj2kd94sqi23kra85mbd7sxpxnwj60pz-pre-commit/bin/pre-commit'
[git.hooks] done
🔨 Welcome to nixos

[devos]

  agenix
  bud                  - Your highly customizable system ctl
  cachix               - Command line client for Nix binary cache hosting https://cachix.org
  deploy-rs            - A Simple multi-profile Nix-flake deploy tool
  nix
  nvfetcher            - Generate nix sources expr for the latest version of packages

[docs]

  mdbook               - Create books from MarkDown

[general commands]

  menu                 - prints this menu

[linter]

  editorconfig-checker - A tool to verify that your files are in harmony with your .editorconfig
  nixpkgs-fmt          - Nix code formatter for nixpkgs

direnv: export +DEVSHELL_DIR +IN_NIX_SHELL +NIXPKGS_PATH +NIX_SSL_CERT_FILE +PRJ_DATA_DIR +PRJ_ROOT ~NIX_PROFILES ~PATH ~XDG_DATA_DIRS

devos  main  ⎔ on  (us-east-1) 
❯ bud update
warning: updating lock file '/tmp/devos/flake.lock':
• Updated input 'agenix':
    'github:ryantm/agenix/e543aa7d68f222e1e771165da9e9a64b5bf7b3e3' (2021-05-13)  → 'github:ryantm/agenix/57806bf7e340f4cae705c91748d4fdf8519293a9' (2021-12-07)• Updated input 'bud':
    'github:divnix/bud/4e709a2fcf19df500ba87d9c3b1b1481401cd53b' (2021-10-04)
  → 'github:divnix/bud/15df091133ff95216c321b09be04905c0a0944c4' (2021-10-10)
• Updated input 'darwin':
    'github:LnL7/nix-darwin/007d700e644ac588ad6668e6439950a5b6e2ff64' (2021-05-26)
  → 'github:LnL7/nix-darwin/44da835ac40dab5fd231298b59d83487382d2fab' (2021-10-23)
• Updated input 'digga':
    'github:divnix/digga/6549cf4c87d1a77138cbecbc6bb9483c20572cd7' (2021-09-29)
  → 'github:divnix/digga/1ce6cef4230f82cb986ef3d08d57a2e3a0e6e80d' (2021-11-29)
• Updated input 'digga/devshell':
    'github:numtide/devshell/7a7a7aa0adebe5488e5abaec688fd9ae0f8ea9c6' (2021-09-23)
  → 'github:numtide/devshell/0e56ef21ba1a717169953122c7415fa6a8cd2618' (2021-11-22)
• Updated input 'digga/flake-utils':
    'github:numtide/flake-utils/f7e004a55b120c02ecb6219596820fcd32ca8772' (2021-06-16)
  → 'github:numtide/flake-utils/74f7e4319258e287b0f9cb95426c9853b282730b' (2021-11-28)
• Updated input 'digga/latest':
    'github:nixos/nixpkgs/31ffc50c571e6683e9ecc9dbcbd4a8e9914b4497' (2021-09-26)  → 'github:nixos/nixpkgs/8a308775674e178495767df90c419425474582a1' (2021-11-29)• Removed input 'digga/nix'
• Removed input 'digga/nix/lowdown-src'
• Removed input 'digga/nix/nixpkgs'
• Updated input 'digga/nixos-generators':
    'github:nix-community/nixos-generators/022ef440af8dc237ab1f59fa363cb1e25783ec3e' (2021-06-29)
  → 'github:nix-community/nixos-generators/05a3eb158a9c7746a5d463726d7f7cccf07500e4' (2021-11-23)
• Updated input 'home':
    'github:nix-community/home-manager/35a24648d155843a4d162de98c17b1afd5db51e4' (2021-06-20)
  → 'github:nix-community/home-manager/7329ffc6e911106494183557fc249180d5422929' (2021-11-30)
• Updated input 'latest':
    'github:nixos/nixpkgs/c464dc811babfe316ed4ab7bbc12351122e69dd7' (2021-08-02)  → 'github:nixos/nixpkgs/581d2d6c9cd5c289002203581d8aa0861963a933' (2021-12-08)• Updated input 'naersk':
    'github:nmattia/naersk/e09c320446c5c2516d430803f7b19f5833781337' (2021-06-17)
  → 'github:nmattia/naersk/ebde51ec0eec82dc71eaca03bc24cf8eb44a3d74' (2021-12-09)
• Updated input 'nixos':
    'github:nixos/nixpkgs/178da37860823d35e801c7df2f73d7866d3d598a' (2021-08-05)  → 'github:nixos/nixpkgs/3b422991781220853035dbe353ba8b7cb279b6b3' (2021-12-10)• Updated input 'nixos-hardware':
    'github:nixos/nixos-hardware/41775780a0b6b32b3d32dcc32bb9bc6df809062d' (2021-07-03)
  → 'github:nixos/nixos-hardware/4c9f07277bd4bc29a051ff2a0ca58c6403e3881a' (2021-12-02)
• Updated input 'nur':
    'github:nix-community/NUR/00c2ec8f0bbdf0cfb2135bde55fbae5d6b64aa6d' (2021-07-15)
  → 'path:/nix/store/6mfkswqi67m35qwv0vh7kpk8rypbl2rq-source?lastModified=1626378135&narHash=sha256-koC6DBYmLCrgXA+AMHVaODf1uHYPmvcFygHfy3eg6vI=&rev=00c2ec8f0bbdf0cfb2135bde55fbae5d6b64aa6d' (2021-07-15)
• Updated input 'nvfetcher':
    'github:berberman/nvfetcher/248fe98ed1bc352ce164d08a9e76fdc5f301ba2d' (2021-09-26)
  → 'github:berberman/nvfetcher/807513f4bbd0e3b5863f4c3b91f8ac846ed6da9b' (2021-10-18)
warning: Git tree '/tmp/devos' is dirty
direnv: loading /tmp/devos/.envrc
direnv: using flake
warning: Git tree '/tmp/devos' is dirty
error: attribute 'patchedNix' missing

       at /nix/store/1bh4cf2rd70wk80c8r0lndq2g8cwn0a8-source/flake.nix:76:15:

           75|             overlays = [
           76|               digga.overlays.patchedNix
             |               ^
           77|               nur.overlay
(use '--show-trace' to show detailed location information)
warning: Git tree '/tmp/devos' is dirty
direnv: renewed cache
direnv: export ~XDG_DATA_DIRS
```

**Comment by codygman** [comment]
Yep, things seem to be fixed with that PR so this should be closed when that is merged in.

**Comment by cipharius** [comment]
I also ran into this problem and manually removed the `patchedNix` overlay.

For me this raises a bigger concern - updating devos flake often leads it to broken state, as it has multiple sperate moving parts. Perhaps devos should pin some of it's most important components to specific tags/versions so that they don't break in case when devos hasn't been adapted to latest upstream changes.

This also applies to nvfetcher, which fails to compile on latest nixos with freshest changes from upstream. Does it really need to be tracking the bleeding edge? I fixed it to tag `4.0.0` and made it track stable nixos branch not latest.
